File: Java_Solutions/LongestConsSequence.java
import java.util.*;

class LongestConsSequence {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> setOfNums = new HashSet<Integer>();
        int maxLengOfSubseq = 0;

        // making a set out of the list
        for (int num : nums) {
            setOfNums.add(num);
        }

        for (Integer num : setOfNums) {
            if (!setOfNums.contains(num - 1)) {
                // previous number to this number is not present in the set
                // this means this number is not part of a sequence rather a starting point for
                // a sequence
                int currSubseqLen = 1;
                int numInSeq = num;

                // now we'll calculate the length of subsequence formed starting with the
                // current element
                while (setOfNums.contains(numInSeq + 1)) {
                    numInSeq += 1;

                    // increase curr subseq length when each subsequent number is present in set
                    currSubseqLen += 1;
                }
                maxLengOfSubseq = Math.max(maxLengOfSubseq, currSubseqLen);
            }
        }
        return maxLengOfSubseq;

    }
}

File: Python_Solutions/LongestConsSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        set_of_nums = set(nums)
        max_len_of_subseq = 0

        for num in set_of_nums:
            if num-1 not in set_of_nums:
                len_of_subseq = 1
                #start of a new sequence
                seq_num = num
                while seq_num+1 in set_of_nums:
                    seq_num += 1
                    len_of_subseq += 1
                max_len_of_subseq = max(max_len_of_subseq,len_of_subseq)
        return max_len_of_subseq